tests/lib/linter/linter.js

```diff
@@ -3134,37 +3134,64 @@ describe("Linter", () => {
 
         it("should report a violation for env changes", () => {
             const code = [
-                `/*${ESLINT_ENV} browser*/`
+                `/*${ESLINT_ENV} browser*/ window`
             ].join("\n");
             const config = {
                 rules: {
-                    test: 2
+                    "no-undef": 2
                 }
             };
-            let ok = false;
+            const messages = linter.verify(code, config, { allowInlineConfig: false });
 
-            linter.defineRules({
-                test(context) {
-                    return {
-                        Program() {
-                            const scope = context.getScope();
-                            const sourceCode = context.getSourceCode();
-                            const comments = sourceCode.getAllComments();
-
-                            assert.strictEqual(1, comments.length);
-
-                            const windowVar = getVariable(scope, "window");
+            assert.strictEqual(messages.length, 1);
+            assert.strictEqual(messages[0].ruleId, "no-undef");
+        });
+    });
 
-                            assert.notOk(windowVar.eslintExplicitGlobal);
+    describe("when evaluating code with 'noInlineComment'", () => {
+        for (const directive of [
+            "globals foo",
+            "global foo",
+            "exported foo",
+            "eslint eqeqeq: error",
+            "eslint-disable eqeqeq",
+            "eslint-disable-line eqeqeq",
+            "eslint-disable-next-line eqeqeq",
+            "eslint-enable eqeqeq",
+            "eslint-env es6"
+        ]) {
+            // eslint-disable-next-line no-loop-func
```

Comment:
Just out of curiosity, why does `no-loop-func` think there is a problem here?

Reply:
Probably due to `CLIEngine` variable. It's shared in the callback in this loop.